include/circt/Dialect/MSFT/MSFTOps.td

```diff
@@ -11,7 +11,10 @@ class MSFTOp<string mnemonic, list<OpTrait> traits = []> :
     Op<MSFTDialect, mnemonic, traits>;
 
 def InstanceOp : MSFTOp<"instance", [
-        DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]> ]> {
+        DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
+        DeclareOpInterfaceMethods<SymbolUserOpInterface>,
+        HasParent<"hw::HWModuleOp">
```

Comment:
Yeah, this is too restrictive. The instances could also live under the upcoming `msft.module`.